From 0bacc0b1118742d60f11969dbe12de698ede593e Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 2 Feb 2016 16:30:50 +0800
Subject: [PATCH 0947/5242] MLK-12371: ARM: imx: suspend-imx7: correct
 HW_ANADIG_SNVS_MISC_CTRL set

commit  2479e8c4a859f1ee00223d8877891736b42a2197 from
https://source.codeaurora.org/external/imx/linux-imx.git

To avoid touch other bits of HW_ANADIG_SNVS_MISC_CTRL , use set/clear register
, and correct the bit29 setting:
  --before: write 1 to toggle DDR power pin to high before enter DDR retention,
            and write 1 again to pull pin to low when exit from DDR retention.
  --now: write 1 to pull DDR power pin to high and write 0 to low.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/suspend-imx7.S |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index 074efdc..4809401 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -79,6 +79,8 @@
 #define GPC_PGC_C0	0x800
 #define GPC_PGC_FM	0xa00
 #define ANADIG_SNVS_MISC_CTRL	0x380
+#define ANADIG_SNVS_MISC_CTRL_SET 0x384
+#define ANADIG_SNVS_MISC_CTRL_CLR 0x388
 #define ANADIG_DIGPROG	0x800
 #define DDRC_STAT	0x4
 #define DDRC_PWRCTL	0x30
@@ -330,11 +332,12 @@
 	ldr	r7, [r11, r6]
 	orr	r7, r7, #0x1
 	str	r7, [r11, r6]
+11:
 	/* turn off ddr power */
 	ldr	r11, [r0, #PM_INFO_MX7_ANATOP_V_OFFSET]
 	ldr	r7, =(0x1 << 29)
-	str	r7, [r11, #ANADIG_SNVS_MISC_CTRL]
-11:
+	str	r7, [r11, #ANADIG_SNVS_MISC_CTRL_SET]
+
 	ldr	r11, [r0, #PM_INFO_MX7_SRC_V_OFFSET]
 	ldr	r6, =0x1000
 	ldr	r7, [r11, r6]
@@ -361,7 +364,7 @@
 
 	/* turn on ddr power */
 	ldr	r7, =(0x1 << 29)
-	str	r7, [r1, #ANADIG_SNVS_MISC_CTRL]
+	str	r7, [r1, #ANADIG_SNVS_MISC_CTRL_CLR]
 
 	ldr	r6, =50
 	wait_delay
@@ -413,7 +416,7 @@
 	str	r7, [r11]
 12:
 	ldr	r7, =(0x1 << 30)
-	str	r7, [r1, #ANADIG_SNVS_MISC_CTRL]
+	str	r7, [r1, #ANADIG_SNVS_MISC_CTRL_SET]
 
 	/* need to delay ~5mS */
 	ldr	r6, =0x100000
-- 
1.7.9.5

